utils/scripts/repo-helper-leap15.sh

```diff
@@ -0,0 +1,97 @@
+# It will also optionally allow running a /tmp/install script
+# for custom packages if present.
+
+: "${REPO_FILE_URL:=}"
+: "${BASE_DISTRO:=opensuse/leap:15.3}"
+: "${JENKINS_URL:=}"
+: "${REPOS:=}"
+
+disable_repo () {
+    if [ -n "$REPO_FILE_URL" ]; then
+        pushd /etc/dnf/repos.d/
+            mv daos_ci-leap15-artifactory.repo{,.tmp}
+            for file in *.repo; do
+                true > "$file"
+            done
+            mv daos_ci-leap15-artifactory.repo{.tmp,}
+        popd
+    fi
+}
+
+# Use local repo server if present
+# if a local repo server is present and the distro repo server can not
+# be reached, have to bootstrap in an environment to get curl installed
+# to then install the pre-built repo file.
+
+if [ -n "$REPO_FILE_URL" ]; then
+    zypper mr --all --disable
+    zypper addrepo                                                            \
+        "${REPO_FILE_URL%/*/}/opensuse-proxy/distribution/leap/${BASE_DISTRO##*:}/repo/oss/" \
+          temp_opensuse_oss_proxy
+    zypper --non-interactive install curl
+    mkdir -p /etc/zypp/repos.d
+    pushd /etc/zypp/repos.d/
+        curl -k -f -o daos_ci-leap15-artifactory.repo.tmp \
+             "$REPO_FILE_URL"daos_ci-leap15-artifactory.repo
+        for file in *.repo; do
+            true > "$file"
+        done
+        mv daos_ci-leap15-artifactory.repo{.tmp,}
+    popd
+fi
+zypper --non-interactive --gpg-auto-import-keys install \
+    dnf dnf-plugins-core
+mkdir -p /etc/dnf/repos.d
+pushd /etc/zypp/repos.d/
+    for file in *.repo; do
+        sed -e '/type=NONE/d' < "$file" > "/etc/dnf/repos.d/$file"
+    done
+popd
+zypper --non-interactive clean --all
+dnf config-manager --save --setopt=assumeyes=True
+dnf config-manager --save --setopt=install_weak_deps=False
+
+daos_base="job/daos-stack/job/"
+artifacts="/artifacts/artifacts/leap15/"
+for repo in $REPOS; do
+    branch="master"
+    build_number="lastSuccessfulBuild"
+    if [[ $repo = *@* ]]; then
+        branch="${repo#*@}"
+        repo="${repo%@*}"
+        if [[ $branch = *:* ]]; then
+            build_number="${branch#*:}"
+            branch="${branch%:*}"
+        fi
+    fi
+    echo -e "[$repo:$branch:$build_number]\n\
+name=$repo:$branch:$build_number\n\
+baseurl=${JENKINS_URL}$daos_base$repo/job/$branch/$build_number$artifacts\n\
+enabled=1\n\
+gpgcheck=False\n" >> /etc/dnf/repos.d/$repo:$branch:$build_number.repo
+    cat /etc/dnf/repos.d/$repo:$branch:$build_number.repo
+done
+
+if [ -e /tmp/install.sh ]; then
+    dnf upgrade
+    disable_repo
+    /tmp/install.sh
+    dnf clean all
+    rm -f /tmp/install.sh
+fi
+
+if [ -e /etc/profile.d/lmod.sh ]; then
+    if ! grep "MODULEPATH=.*/usr/share/modules" /etc/profile.d/lmod.sh; then
+        sed -e '/MODULEPATH=/s/$/:\/usr\/share\/modules/' \
+               /etc/profile.d/lmod.sh
+    fi
+fi
+
+# this should not be needed, but docker containers have for leap 15
+# have failed in the past if it is not present.
```

Comment:
Grammar error.  Are you saying they intermittently fail or that we're just not sure if this is needed or not?

Reply:
Yes, in the past leap 15 dockerfile builds would start fail to validate https certificates for public web sites.
It was traced down to that the package for the updated certificates was installed, but for some unknown reason, those certificates where not deployed to the application specific directories which is what the command below does.
We could not reproduce the issue outside of a docker container.
So this is carried forward just in case the issue shows up again.

I will fix the grammar error.